### PR TITLE
Replace numbers with HTTP status code

### PR DESCRIPTION
**Pull request by ChenglongGao327**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/model/jwks_resolver.go [hunk 1]
```diff
@@ -345,7 +345,7 @@ func (r *JwksResolver) getRemoteContentWithRetry(uri string, retry int) ([]byte,
 			return nil, err
 		}
 
-		if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		if resp.StatusCode < http.StatusOK || resp.StatusCode >= http.StatusMultipleChoices {
```
Comment (ramaraochavali) [inline pilot/pkg/model/jwks_resolver.go:348]: In such comparisons I think having numbers may be more clear - this is my personal preference. But let us wait for others feedback as well.

Comment (ChenglongGao327) [reply pilot/pkg/model/jwks_resolver.go:348]: > In such comparisons I think having numbers may be more clear - this is my personal preference. But let us wait for others feedback as well.


I agree with you ，let me recovert it

File: security/pkg/nodeagent/plugin/providers/google/stsclient/stsclient.go [hunk 2]
```diff
@@ -74,7 +74,10 @@ func NewSecureTokenServiceExchanger(credFetcher security.CredFetcher, trustDomai
 }
 
 func retryable(code int) bool {
-	return code >= 500 && !(code == 501 || code == 505 || code == 511)
+	return code >= http.StatusInternalServerError &&
```
Comment (ramaraochavali) [inline security/pkg/nodeagent/plugin/providers/google/stsclient/stsclient.go:77]: here also please change to code >= 500

File: pkg/wasm/httpfetcher.go [hunk 3]
```diff
@@ -78,5 +78,8 @@ func (f *HTTPFetcher) Fetch(url string, timeout time.Duration) ([]byte, error) {
 }
 
 func retryable(code int) bool {
-	return code >= 500 && !(code == 501 || code == 505 || code == 511)
+	return code >= http.StatusInternalServerError &&
```
Comment (ramaraochavali) [inline pkg/wasm/httpfetcher.go:81]: here also please change to code >= 500

Comment (ChenglongGao327) [reply pkg/wasm/httpfetcher.go]: > here also please change to code >= 500

done

**Review by ramaraochavali — APPROVED** [review]

**Review by bianpengyuan — APPROVED** [review]

**Review by morvencao — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**